tools/codemods/src/codemods/consolidate-popover-usePortal-renderMode-props/transform.ts

```diff
@@ -0,0 +1,64 @@
+import type { API, FileInfo } from 'jscodeshift';
+
+import {
+  consolidateJSXAttributes,
+  ConsolidateJSXAttributesOptions,
+} from '../../utils/transformations';
+import { addJSXAttributes } from '../../utils/transformations/addJSXAttributes';
+
+type TransformerOptions = ConsolidateJSXAttributesOptions & {
+  componentName: string;
+};
+
+/**
+ * Example transformer function to consolidate props
+ *
+ * @param file the file to transform
+ * @param jscodeshiftOptions an object containing at least a reference to the jscodeshift library
+ * @param options an object containing options to pass to the transform function
+ * @returns Either the modified file or the original file
+ */
+export default function transformer(
+  file: FileInfo,
+  { jscodeshift: j }: API,
+  options: TransformerOptions,
+) {
+  const source = j(file.source);
+
+  const {
+    propToRemove = 'usePortal',
+    propToUpdate = 'renderMode',
+    propMapping = {
+      undefined: 'portal',
+      false: 'inline',
+      true: 'portal',
+    },
+    propToRemoveType = 'boolean',
+    componentName = 'Popover',
+  } = options;
+
+  // Check if the element is on the page
+  const elements = source.findJSXElements(componentName);
+
+  // If there are no elements then return the original file
+  if (elements.length === 0) return file.source;
+
+  elements.forEach(element => {
+    addJSXAttributes({
```

Comment:
Is this adding the `usePortal` prop if it doesn't exist and then updating it to `renderMode` with `consolidateJSXAttributes`?